libraries/Hosting/microsoft-agents-hosting-aiohttp/microsoft/agents/hosting/aiohttp/app/streaming/streaming_response.py

```diff
@@ -0,0 +1,382 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import asyncio
+import logging
+from typing import List, Optional, Callable, Literal, TYPE_CHECKING
+from dataclasses import dataclass
+
+from microsoft.agents.core.models import (
+    Activity,
+    Entity,
+    Attachment,
+    ClientCitation,
+    SensitivityUsageInfo,
+    add_ai_to_activity,
+)
+
+if TYPE_CHECKING:
+    from microsoft.agents.builder.turn_context import TurnContext
+
+from .citation import Citation
+from .citation_util import CitationUtil
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class StreamingChannelData:
+    """
+    Structure of the outgoing channelData field for streaming responses.
+
+    The expected sequence of streamTypes is:
+    'informative', 'streaming', 'streaming', ..., 'final'.
+
+    Once a 'final' message is sent, the stream is considered ended.
+    """
+
+    stream_type: Literal["informative", "streaming", "final"]
+    """The type of message being sent."""
+
+    stream_sequence: int
+    """Sequence number of the message in the stream. Starts at 1 for the first message."""
+
+    stream_id: Optional[str] = None
+    """ID of the stream. Assigned after the initial update is sent."""
+
+
+class StreamingResponse:
+    """
+    A helper class for streaming responses to the client.
+
+    This class is used to send a series of updates to the client in a single response.
+    The expected sequence of calls is:
+
+    `queue_informative_update()`, `queue_text_chunk()`, `queue_text_chunk()`, ..., `end_stream()`.
+
+    Once `end_stream()` is called, the stream is considered ended and no further updates can be sent.
+    """
+
+    def __init__(self, context: "TurnContext"):
+        """
+        Creates a new StreamingResponse instance.
+
+        Args:
+            context: Context for the current turn of conversation with the user.
+        """
+        self._context = context
+        self._next_sequence = 1
+        self._stream_id: Optional[str] = None
+        self._message = ""
+        self._attachments: Optional[List[Attachment]] = None
```

Comment:
Attachments were replaced with an Activity, per Ickman request.  I believe the same thing can happen though, which is the final message can contain an Adaptive Card (for example).